tests/test_completion.py

```diff
@@ -35,3 +38,147 @@ def test_completion_model_create_override(mock_openai_credentials: MagicMock):
+        return 1 / divisor
+
+    def test_sync_fn(self, mock_openai_credentials):
+        with pytest.raises(ZeroDivisionError, match="\nOpenAI"):
+            self.sync_fn(0)
+        mock_openai_credentials._mock_block_load.assert_called_once_with("curie")
+        assert mock_openai_credentials._mock_block_load.call_count == 1
+
+    @pytest.mark.parametrize("return_state", [True, False])
+    def test_sync_flow(self, mock_openai_credentials, return_state):
+        with pytest.raises(ZeroDivisionError, match="\nOpenAI"):
+            result = flow(self.sync_fn)(0, return_state=return_state)
+            if return_state:
+                result = result.result()
+        mock_openai_credentials._mock_block_load.assert_called_once_with("curie")
+        assert mock_openai_credentials._mock_block_load.call_count == 1
+
+    @pytest.mark.parametrize("return_state", [True, False])
+    def test_sync_task(self, mock_openai_credentials, return_state):
+        @flow
+        def a_flow():
+            return task(self.sync_fn)(0, return_state=return_state)
+
+        with pytest.raises(ZeroDivisionError, match="\nOpenAI"):
+            result = a_flow()
+            if return_state:
+                result = result.result()
+        mock_openai_credentials._mock_block_load.assert_called_once_with("curie")
+        assert mock_openai_credentials._mock_block_load.call_count == 1
+
+    @interpret_exception("curie")
+    async def async_fn(self, divisor: int):
+        return 1 / divisor
+
+    async def test_async_fn(self, mock_openai_credentials):
+        with pytest.raises(ZeroDivisionError, match="\nOpenAI"):
+            await self.async_fn(0)
+        mock_openai_credentials._mock_block_load.assert_called_once_with("curie")
+        assert mock_openai_credentials._mock_block_load.call_count == 1
+
+    @pytest.mark.parametrize("return_state", [True, False])
+    async def test_async_flow(self, mock_openai_credentials, return_state):
+        with pytest.raises(ZeroDivisionError, match="\nOpenAI"):
+            result = await (flow(self.async_fn)(0, return_state=return_state))
+            if return_state:
+                result = await result.result(fetch=True)
+        mock_openai_credentials._mock_block_load.assert_called_once_with("curie")
+        assert mock_openai_credentials._mock_block_load.call_count == 1
+
+    @pytest.mark.parametrize("return_state", [True, False])
+    async def test_async_task(self, mock_openai_credentials, return_state):
+        @flow
+        async def a_flow():
+            return await task(self.async_fn)(0, return_state=return_state)
+
+        with pytest.raises(ZeroDivisionError, match="\nOpenAI"):
+            result = await a_flow()
+            if return_state:
+                result = await result.result(fetch=True)
+        mock_openai_credentials._mock_block_load.assert_called_once_with("curie")
+        assert mock_openai_credentials._mock_block_load.call_count == 1
+
```

Comment:
what do you think about a test that ensures task/flow decorator kwargs are passed through?